### PR TITLE
Add hddtemp check script

### PR DESCRIPTION
**Pull request by SlothOfAnarchy**

### CODE REVIEW

File: files/check_hddtemp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        elif temp >= args['temp_warn']:
+            raiseState('WARNING', ret)
+        max_temp = max(max_temp, temp)
+        min_temp = min(min_temp, temp)
+        ret['perfdata'] += '{}={}°C;{};{};0;65 '.format(dev,temp,args['temp_warn'],args['temp_crit'])
+        ret['details'] += '{} ({}): {}°C\n'.format(dev, model, temp)
+
+    ret['output'] = 'Maximum HDD temperature is in range [{}, {}].'.format(min_temp, max_temp)
+
+doCheck(check)
```
Comment (Mr-Pi) [inline files/check_hddtemp]: Please add final newline
```suggestion
doCheck(check)

```

File: files/check_hddtemp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            if not os.access(dev_path, os.R_OK):
+                raiseState('UNKNOWN', ret)
+                ret['output'] = 'Not allowed to access {}'.format(dev_path)
+                return
+            scan_devices.append(dev_path)
+
+    hddtemp_process = subprocess.Popen(hddtemp_command + scan_devices, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    hddtemp_output = hddtemp_process.communicate()[0].decode('ISO-8859-1')
+
+    min_temp = 65
```
Comment (Mr-Pi) [inline files/check_hddtemp]: Can you please describe why the `min_temp` and `max_temp` are required and the usage of those?

**Review by Mr-Pi — CHANGES_REQUESTED** [review]

File: files/check_hddtemp (hunk 2 above)
Comment (SlothOfAnarchy) [reply files/check_hddtemp]: They're used in the final output of the check: https://github.com/stuvusIT/icinga2-scripts/pull/23/files#diff-aea7ff42554e3991b1b0e15f6777f114R88
Without these, the message would just be that the check is okay, but that information is already provided by the check status. Printing the temperature for all devices there would be cluttered and have a differing length depending on the number of devices. That's why I chose to calculate and display the coldest and hottest temperature.

File: files/check_hddtemp [hunk 3]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
```
Comment (dasJ) [inline files/check_hddtemp]: Can you add a comment at the beginning describing what the check checks and how it's called? I know it's documented further down but I think it's a lot faster to just have it on top. Maybe just paste `--help` there. Also you can get rid of that emacs garbage there

File: files/check_hddtemp [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                        default=40)
+    parser.add_argument('-tc', '--temp-crit',
+                        help="Return CRITICAL if the temperature in degree Celcius of any device is higher than this", type=int,
+                        default=50)
+    parser.add_argument('-i', '--ignore',
+                        help="Ignore this device, e.g. 'sda'. May be provided multiple times.", default=[], action='append')
+    args = vars(parser.parse_args())
+
+    lsblk_process = subprocess.Popen(['lsblk', '-SJ', '--output=NAME,MODEL'], stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    lsblk_json_output = lsblk_process.communicate()[0].decode('ISO-8859-1')
```
Comment (dasJ) [inline files/check_hddtemp:55]: Are you sure that's not utf-8?

File: files/check_hddtemp [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            dev_path = '/dev/{}'.format(device['name'])
+            # Check if access to device is allowed for the executing user
+            if not os.access(dev_path, os.R_OK):
+                raiseState('UNKNOWN', ret)
+                ret['output'] = 'Not allowed to access {}'.format(dev_path)
+                return
+            scan_devices.append(dev_path)
+
+    hddtemp_process = subprocess.Popen(hddtemp_command + scan_devices, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    hddtemp_output = hddtemp_process.communicate()[0].decode('ISO-8859-1')
```
Comment (dasJ) [inline files/check_hddtemp:85]: Again, what if this is utf-8?

File: files/check_hddtemp (hunk 2 above)
Comment (dasJ) [reply files/check_hddtemp]: Why 65 and not something like 999?

**Review by dasJ — CHANGES_REQUESTED** [review]

**Comment by dasJ** [comment]
Also:

```
$ flake8 check_hddtemp
check_hddtemp:14:1: E302 expected 2 blank lines, found 1
check_hddtemp:28:80: E501 line too long (127 > 79 characters)
check_hddtemp:31:80: E501 line too long (128 > 79 characters)
check_hddtemp:34:80: E501 line too long (124 > 79 characters)
check_hddtemp:37:80: E501 line too long (148 > 79 characters)
check_hddtemp:42:80: E501 line too long (118 > 79 characters)
check_hddtemp:66:80: E501 line too long (141 > 79 characters)
check_hddtemp:71:12: E231 missing whitespace after ','
check_hddtemp:85:60: E231 missing whitespace after ','
check_hddtemp:85:65: E231 missing whitespace after ','
check_hddtemp:85:80: E501 line too long (101 > 79 characters)
check_hddtemp:85:83: E231 missing whitespace after ','
check_hddtemp:88:80: E501 line too long (86 > 79 characters)
check_hddtemp:90:1: E305 expected 2 blank lines after class or function definition, found 1
```

**Review by haslersn — CHANGES_REQUESTED** [review]
Proposed change: [check-hddtemp-vol2](/stuvusIT/icinga2-scripts/compare/check-hddtemp-vol2)

**Review by dasJ — DISMISSED** [review]

File: files/check_hddtemp (hunk 2 above)
Comment (SlothOfAnarchy) [reply files/check_hddtemp]: I think 65 is a good value for a maximum temperatur a HDD can have. If you choose to much, the pie diagrams in icingaweb2 will show mostly the same value.
I added an option to set the maximum value, though.

File: files/check_hddtemp (hunk 4 above)
Comment (SlothOfAnarchy) [reply files/check_hddtemp:55]: Yes, I tried that first, but it doesn't work for all devices, e.g. my local SSD is not in the database and outputs characters not included in `UTF-8`.

**Comment by haslersn** [comment]
With your version I get the output:

```
hddtemp UNKNOWN: Python exception occured
Traceback (most recent call last):
  File "/home/haslersn/projects/mr-pi/stuvus_config/roles/icinga2-scripts/files/common.py", line 60, in doCheck
    checkFunc(ret)
  File "files/check_hddtemp", line 111, in check
    temp = int(line)
ValueError: invalid literal for int() with base 10: '/dev/sr0: HL-DT-ST DVDRAM GUE1N                   : S.M.A.R.T. not available'
```

EDIT: [check-hddtemp-vol2](/stuvusIT/icinga2-scripts/compare/check-hddtemp-vol2) yields
```
hddtemp UNKNOWN: HDD temperature is in range [29, 36]. | sda=29°C;40;50;0;null sdb=36°C;40;50;0;null
sda (CT480BX200SSD1): 29°C
sdb (HFS512G39TND-N210A): 36°C
sr0 (HL-DT-ST_DVDRAM_GUE1N): /dev/sr0: HL-DT-ST DVDRAM GUE1N                   : S.M.A.R.T. not available
```

**Comment by SlothOfAnarchy** [comment]
@haslersn fixed

**Pull request closed without merging**